dimod/utilities.py

```diff
@@ -340,3 +345,48 @@ def resolve_label_conflict(mapping, old_labels=None, new_labels=None):
             # don't need to add it to intermediate_to_new because it is a self-label
 
     return old_to_intermediate, intermediate_to_new
+
+
+def child_structure_dfs(sampler, seen=None):
+    """Find the structure of a composed sampler by a depth first search on the
+    children.
+
+    Args:
+        sampler (:obj:`.Sampler`):
+            A :class:`.Structured` sampler or a composed sampler with at least
+            one structured child.
+
+        seen (set, optional, default=False):
+            A set of the ids of the child samplers that have already been
+            checked.
+
+    Returns:
+        :class:`~collections.namedtuple`: A named tuple of the form
+        `Structure(nodelist, edgelist, adjacency)`, where the 3-tuple values
+        are the :attr:`.Structured.nodelist`, :attr:`.Structured.edgelist`
+        and :attr:`.Structured.adjacency` attributes of the first structured
+        sampler encountered.
+
+    """
+    seen = set() if seen is None else seen
+
+    if id(sampler) not in seen:
```

Comment:
`id` not needed, you could just add `sampler` object to the set -- that's cheap, only reference is added. And membership testing is just as cheap, since sampler's hashes are equal to their `id`s already.

Reply:
You're right, for some reason I thought it would not be hashable.